File: go/domain/product.go
// ABOUTME: Product domain type representing a product in the catalog.
// ABOUTME: Contains product identification, name, and categorization.
package domain

// Product represents a product in the catalog.
type Product struct {
	Sku      Sku
	Name     string
	Category string
}

// NewProduct creates a new Product.
func NewProduct(sku Sku, name, category string) Product {
	return Product{
		Sku:      sku,
		Name:     name,
		Category: category,
	}
}

File: typescript/src/domain/Product.ts
// ABOUTME: Product catalog entry with SKU, name, and category
// ABOUTME: Immutable domain model representing a product in the catalog

import { Sku } from './Sku';

export class Product {
  constructor(
    public readonly sku: Sku,
    public readonly name: string,
    public readonly category: string
  ) {}
}
